python/neutron-understack/neutron_understack/neutron_understack_mech.py

```diff
@@ -300,45 +277,33 @@ def check_segment(self, segment):
     def check_vlan_transparency(self, context):
         log_call("check_vlan_transparency", context)
 
-    def _move_to_network(
-        self,
-        vif_type: str,
-        mac_address: str,
-        device_uuid: UUID,
-        network_id: str,
-        argo_client: ArgoClient,
-    ):
-        """Triggers Argo "trigger-undersync" workflow.
-
-        This has the effect of connecting our server to the given networks: either
-        "provisioning" (for PXE booting) or "tenant" (normal access to customer's
-        networks).  The choice of network is based on the network ID.
-
-        This only happens when vif_type is VIF_TYPE_OTHER.
-
-        argo_client is injected by the caller to make testing easier
-        """
-        if vif_type != portbindings.VIF_TYPE_OTHER:
+    def fetch_connected_interface_uuid(self, context):
+        connected_interface_uuid = (
+            context.current["binding:profile"]
+            .get("local_link_information")[0]
+            .get("port_id")
+        )
+        try:
+            UUID(str(connected_interface_uuid))
+        except ValueError:
+            LOG.debug(
+                "Local link information port_id is not a valid UUID type"
+                " port_id: %(connected_interface_uuid)s",
+                {"connected_interface_uuid": connected_interface_uuid},
+            )
             return
+        return connected_interface_uuid
 
+    def update_nautobot(self, network_id, connected_interface_uuid):
         if network_id == cfg.CONF.ml2_type_understack.provisioning_network:
-            network_name = "provisioning"
+            port_status = "Provisioning-Interface"
```

Comment:
So this sets the port status to Provisioning, doesn't it have to set it back again afterwards?

Reply:
this will be handled either in port delete call(not implemented yet) or it is already handled when flipping port to tenant network in prep_switch_interface job